tests/tcp.rs

```diff
@@ -1134,3 +1134,31 @@ fn socket_to_nonexistent_node() -> Result {
     });
     sim.run()
 }
+
+#[should_panic]
+#[test]
+fn exhaust_ephemeral_ports() {
+    let mut sim = Builder::new()
+        .tick_duration(Duration::from_millis(1))
+        .simulation_duration(Duration::from_secs(60))
+        .build();
```

Comment:
If the range was configurable, this test could be "faster" if you gave it a smaller range.